### PR TITLE
Add SendGrid Marketing Campaigns Source docs

### PR DESCRIPTION
**Pull request by FirstWhack**
### Proposed changes

Adds a new Source doc for Sendgrid Marketing Campaigns

### Merge timing
Release date is scheduled for Monday June 17

### Related issues (optional)
https://github.com/twilio-internal/docs/pull/1721

### CODE REVIEW

**Comment by FirstWhack** [comment]
I'm confused by the syntax at the very top of the template (with `title`), looking for some help understanding if this is right

File: src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
```
Comment (FirstWhack) [inline src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md:1]: What is this syntax?

File: src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+title: [SendGrid Marketing Campaigns] Source
+---
+
+[SendGrid Marketing Campaigns](https://yourintegration.com/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank”} enables you to automatically stream your email events directly into Segment to be used inside your warehouse or other downstream destinations.
```
Comment (FirstWhack) [inline src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md:5]: This link is not yet available and needs to be updated before merging

File: src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+---
+title: [SendGrid Marketing Campaigns] Source
+---
```
Comment (tcgilbert) [inline src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md:3]:
```suggestion
---
title: SendGrid Marketing Campaigns Source
id: GCeG0vmcDW
beta: true
---
```

File: src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md (hunk 1 above)
Comment (tcgilbert) [reply src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md:1]: frontmatter for markdown: https://mystmd.org/guide/frontmatter

File: src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+This is an [Event Cloud Source](/docs/sources/#event-cloud-sources) which can not only export data into your Segment warehouse, but can also federate the exported data into your other enabled Segment Destinations.
+
+This source is maintained by SendGrid Marketing Campaigns. For any issues with the source, [contact their Support team](mailto:support@sendgrid.com).
+
+## Getting started
+1. From your workspace's [Sources catalog page](https://app.segment.com/goto-my-workspace/sources/catalog){:target="_blank”} click **Add Source**.
+2. Search for "SendGrid Marketing Campaigns" in the Sources Catalog, select SendGrid Marketing Campaigns, and click **Add Source**.
+3. On the next screen, give the Source a name configure any other settings.
+   - The name is used as a label in the Segment app, and Segment creates a related schema name in your warehouse. The name can be anything, but we recommend using something that reflects the source itself and distinguishes amongst your environments (eg. SourceName_Prod, SourceName_Staging, SourceName_Dev).
```
Comment (tcgilbert) [inline src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md:15]:
```suggestion
   - The name is used as a label in the Segment app, and Segment creates a related schema name in your warehouse. The name can be anything, but we recommend using something that reflects the source itself and distinguishes amongst your environments (eg. Sendgrid_Prod, Sendgrid_Staging, Sendgrid_Dev).
```

File: src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+| Email Link Clicked      | Recipient clicked the [tracking link](https://www.twilio.com/docs/sendgrid/ui/account-and-settings/tracking#click-tracking) |
+| Email Dropped           | Email was [dropped by Sendgrid](https://www.twilio.com/docs/sendgrid/glossary/drops)                                        |
+| Email Bounced           | Email servers [rejected the email](https://www.twilio.com/docs/sendgrid/ui/sending-email/bounces)                           |
+| Unsubscribed            | Recipient [unsubscribed globally](https://www.twilio.com/docs/sendgrid/ui/sending-email/global-unsubscribes)                |
+| Unsubscribed From Group | Recipient [unsubscribed from a group](https://www.twilio.com/docs/sendgrid/ui/sending-email/group-unsubscribes)             |
+| Resubscribed To Group   | Recipient resubscribed to a group                                                                                           |
+| Email Marked as Spam    | Recipient [reported the email as spam](https://www.twilio.com/docs/sendgrid/ui/analytics-and-reporting/spam-reports)        |
```
Comment (tcgilbert) [inline src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md:42]:
```suggestion
| Email Machine Opened    | Email was [opened anonymously](https://www.twilio.com/docs/sendgrid/for-developers/tracking-events/understanding-apple-mail-privacy-protection-and-open-events){:target="_blank”} |
| Email Link Clicked      | Recipient clicked the [tracking link](https://www.twilio.com/docs/sendgrid/ui/account-and-settings/tracking#click-tracking){:target="_blank”} |
| Email Dropped           | Email was [dropped by Sendgrid](https://www.twilio.com/docs/sendgrid/glossary/drops){:target="_blank”}                                        |
| Email Bounced           | Email servers [rejected the email](https://www.twilio.com/docs/sendgrid/ui/sending-email/bounces){:target="_blank”}                           |
| Unsubscribed            | Recipient [unsubscribed globally](https://www.twilio.com/docs/sendgrid/ui/sending-email/global-unsubscribes){:target="_blank”}                |
| Unsubscribed From Group | Recipient [unsubscribed from a group](https://www.twilio.com/docs/sendgrid/ui/sending-email/group-unsubscribes){:target="_blank”}             |
| Resubscribed To Group   | Recipient resubscribed to a group                                                                                           |
| Email Marked as Spam    | Recipient [reported the email as spam](https://www.twilio.com/docs/sendgrid/ui/analytics-and-reporting/spam-reports){:target="_blank”}        |
```

File: src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+| `email_type`                | "Singlesend" or "Automation"                                                                                                    |
+| `template_id`               | Email template ID                                                                                                               |
+| `email_subject`             | Subject line of the email, when available                                                                                       |
+| `test_phase`                | `true` if the Single Send is an [A/B Test](https://www.twilio.com/docs/sendgrid/ui/sending-email/a-b-testing) in the test phase |
+| `unsubscribe_group_id`      | ID of the unsubscribe group, when available                                                                                     |
+| `link_url`                  | URL of the link clicked                                                                                                         |
+| `categories`                | [Categories](https://www.twilio.com/docs/sendgrid/for-developers/sending-email/categories) associated to the email              |
```
Comment (tcgilbert) [inline src/connections/sources/catalog/cloud-apps/sendgrid-marketing-campaigns/index.md:57]:
```suggestion
| `test_phase`                | `true` if the Single Send is an [A/B Test](https://www.twilio.com/docs/sendgrid/ui/sending-email/a-b-testing){:target="_blank”} in the test phase |
| `unsubscribe_group_id`      | ID of the unsubscribe group, when available                                                                                     |
| `link_url`                  | URL of the link clicked                                                                                                         |
| `categories`                | [Categories](https://www.twilio.com/docs/sendgrid/for-developers/sending-email/categories){:target="_blank”} associated to the email              |
```

**Review by tcgilbert — CHANGES_REQUESTED** [review]

**Review by tcgilbert — APPROVED** [review]

**Comment by tcgilbert** [comment]
@cmastr if we can get this included in tuesday's deploy that would be great!

**Review by pwseg — APPROVED** [review]

**Comment by pwseg** [comment]
>  if we can get this included in tuesday's deploy that would be great!

Merged!  It'll go out on Tuesday.

**Pull request merged**